src/sentry/uptime/config_producer.py

```diff
@@ -62,3 +73,53 @@ def _produce_to_kafka(
     )
     result = _configs_producer.produce(ArroyoTopic(topic), payload)
     result.result()
+
+
+def get_partition_from_subscription_id(subscription_id: UUID) -> int:
+    return int(subscription_id) % settings.UPTIME_CONFIG_PARTITIONS
+
+
+def get_partition_keys(subscription_id: UUID) -> tuple[str, str]:
+    partition = get_partition_from_subscription_id(subscription_id)
+    return f"uptime:configs:{partition}", f"uptime:updates:{partition}"
+
+
+def _send_to_redis(
+    destination_region_slug: str, subscription_id: UUID, value: bytes | None
+) -> None:
+    region_config = get_region_config(destination_region_slug)
+    if region_config is None:
+        logger.error(
+            "Attempted to create uptime subscription with invalid region slug",
+            extra={"region_slug": destination_region_slug, "subscription_id": subscription_id},
+        )
+        return
+
+    partition = get_partition_from_subscription_id(subscription_id)
+    key = subscription_id.hex
+
+    config_key = f"uptime:configs:{partition}"
+    update_key = f"uptime:updates:{partition}"
+
+    cluster: RedisCluster | StrictRedis = redis.redis_clusters.get_binary(
+        region_config.config_redis_cluster
+    )
+    pipe = cluster.pipeline()
+    if value is None:
+        pipe.hdel(config_key, key)
+        action = "delete"
```

Comment:
Would be nice to have a type for this, maybe a dataclass that knows how to serialize itself


```py
@dataclass
class ConfigUpdate():
  action: Literal["delete" | "upsert"]
  subscription_id: Uuid

  def as_msgpack(self):
      return msgpack.packb(
          {
              "action": self.action,
              "subscription_id": self.subscription_id.hex,
          }
      )
```

Reply:
potentially could be redis aware as well

```py
@property
def redis_update_key(self):
  partition = get_partition_from_subscription_id(self.subscription_id)
  return f"uptime:updates:{partition}"
```

Just a thought